lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,327 @@
+          selectedUserIds: selectedUserIds),
+        Expanded(child: NewDmUserList(
+          filteredUsers: filteredUsers,
+          selectedUserIds: selectedUserIds,
+          onUserSelected: (userId) {
+            setState(() {
+              if (selectedUserIds.contains(userId)) {
+                selectedUserIds.remove(userId);
+              } else {
+                selectedUserIds.add(userId);
+              }
+              scrollToSearchField();
+            });
+          }))]));
+  }
+}
+
+class NewDmHeader extends StatelessWidget {
+  const NewDmHeader({
+    super.key,
+    required this.selectedUserIds,
+    required this.onClose});
+
+  final Set<int> selectedUserIds;
+  final VoidCallback onClose;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    return Container(
+      padding: const EdgeInsets.all(8),
+      child: Row(
+        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+        children: [
+          _buildBackButton(context, designVariables, zulipLocalizations),
+          Text(zulipLocalizations.newDmSheetScreenTitle,
+            style: TextStyle(color: designVariables.title,fontSize: 20, fontWeight: FontWeight.w600)),
+          _buildNextButton(context, store, designVariables, zulipLocalizations)]));
+  }
+
+  Widget _buildBackButton(BuildContext context, DesignVariables designVariables,
+      ZulipLocalizations zulipLocalizations) => TextButton(
+    onPressed: onClose,
+    child: Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Icon(ZulipIcons.chevron_left, size: 24, color: designVariables.icon),
+        Text(zulipLocalizations.newDmSheetBackButtonLabel,
+          style: TextStyle(fontSize: 20, color: designVariables.icon))]));
+
+  Widget _buildNextButton(BuildContext context, PerAccountStore store,
+      DesignVariables designVariables, ZulipLocalizations zulipLocalizations) => TextButton(
+    onPressed: selectedUserIds.isEmpty ? null : () {
+      final narrow = DmNarrow.withOtherUsers(
+        selectedUserIds,
+        selfUserId: store.selfUserId);
+      Navigator.pop(context);
+      Navigator.push(context, MessageListPage.buildRoute(
+        context: context,
+        narrow: narrow));
+    },
+    child: Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Text(zulipLocalizations.newDmSheetNextButtonLabel,
+          style: TextStyle(
+            fontSize: 20,
+            fontWeight: FontWeight.w600,
+            color: selectedUserIds.isEmpty
+              ? designVariables.icon.withFadedAlpha(0.5)
+              : designVariables.icon)),
+        Icon(ZulipIcons.chevron_right,
+          size: 24,
+          color: selectedUserIds.isEmpty
+            ? designVariables.icon.withFadedAlpha(0.5)
+            : designVariables.icon)]));
```

Comment:
Square brackets are usually not wrapped, because they help indicate the end of a list and that the items before them are parallel.